app/rdf/query-latest-cube-iri.ts

```diff
@@ -0,0 +1,44 @@
+/** Creates SPARQL query to fetch latest cube iri.
+ * Works for both versioned and unversioned cubes.
+ */
+export const getLatestCubeIriQuery = (cubeIri: string) => {
+  return `PREFIX cube: <https://cube.link/>
+PREFIX schema: <http://schema.org/>
+
+SELECT ?iri WHERE {
+  {
+    # Versioned cube.
+    SELECT ?iri WHERE {
+      VALUES ?oldIri { <${cubeIri}> }
+      ?versionHistory schema:hasPart ?oldIri .
+      ?versionHistory schema:hasPart ?iri .
+      ?iri schema:version ?version .
+      ?iri schema:creativeWorkStatus ?status .
+      ?oldIri schema:creativeWorkStatus ?oldStatus .
+      FILTER(NOT EXISTS { ?iri schema:expires ?expires . } && ?status IN (?oldStatus, <https://ld.admin.ch/vocabulary/CreativeWorkStatus/Published>))
+    }
+  } UNION {
+    {
+      # Non-versioned cube.
+      SELECT ?iri WHERE {
+        VALUES ?iri { <${cubeIri}> }
+        ?iri cube:observationConstraint ?shape .
+        ?iri schema:creativeWorkStatus ?status .
+        FILTER(NOT EXISTS { ?iri schema:expires ?expires . } && ?status = <https://ld.admin.ch/vocabulary/CreativeWorkStatus/Published>)      
+      }
+    }
+  } UNION {
+    {
+      # Version history of a cube.
+      VALUES ?versionHistory { <${cubeIri}> }
+      ?versionHistory schema:hasPart ?iri .
+      ?iri schema:version ?version .
+      ?iri schema:creativeWorkStatus ?status .
+      ?oldIri schema:creativeWorkStatus ?oldStatus .
+      FILTER(NOT EXISTS { ?iri schema:expires ?expires . } && ?status IN (?oldStatus, <https://ld.admin.ch/vocabulary/CreativeWorkStatus/Published>))
+    }
```

Comment:
Sure, with this part we also catch cases like `https://culture.ld.admin.ch/sfa/StateAccounts_Function`, where we specify like to a cube's version history :)

We had this defined in one of the E2E tests, I think it might have been used in some links from OpenData.swiss datasets 👀

Reply:
Good catch, thanks! I corrected this 👍